python/cuml/decomposition/pca.pyx

```diff
@@ -479,8 +479,8 @@ class PCA(Base):
         X_m, trans_input_ptr, n_rows, _, dtype = \
             input_to_dev_array(X, check_dtype=self.dtype,
                                convert_to_dtype=(self.dtype if convert_dtype
-                                                 else None),
-                               check_cols=self.n_cols)
+                                                 else None)
```

Comment:
why this change? We need to check the number of columns here, no?

Reply:
in inverse transform the transformed dataset might have less number of cols than the original dataset and if we check the number of cols here then it will compare it with the original dataset and throw an error.

Reply:
ah very true, my bad :)